Tests/CheckForUpdates.Action.Test.ps1

```diff
@@ -1,949 +1,2306 @@
+
+        $customJobs[0].Name | Should -Be 'CustomJob-MyFinalJob'
+        $customJobs[0].Origin | Should -Be 'FinalRepository'
+
+        $customJobs[1].Name | Should -Be 'CustomJob-MyCustomTemplateJob'
+        $customJobs[1].Origin | Should -Be 'TemplateRepository'
+
+        $emptyCustomJobs = $nonCustomizedYaml.GetCustomJobsFromYaml('CustomJob*')
+        $emptyCustomJobs | Should -BeNullOrEmpty
+    }
+
+    It 'Test YamlClass AddCustomJobsToYaml' {
+        . (Join-Path $scriptRoot "yamlclass.ps1")
+
+        $customTemplateYaml = [Yaml]::load((Join-Path $PSScriptRoot 'CustomizedYamlSnippet-TemplateRepository.txt'))
+        $finalRepositoryYaml = [Yaml]::load((Join-Path $PSScriptRoot 'CustomizedYamlSnippet-FinalRepository.txt'))
+        $nonCustomizedYaml = [Yaml]::load((Join-Path $PSScriptRoot 'YamlSnippet.txt'))
+
+        $customTemplateJobs = $customTemplateYaml.GetCustomJobsFromYaml('CustomJob*')
+        $customTemplateJobs | Should -Not -BeNullOrEmpty
+        $customTemplateJobs.Count | Should -be 1
+        $customTemplateJobs[0].Name | Should -Be 'CustomJob-MyCustomTemplateJob'
+        $customTemplateJobs[0].Origin | Should -Be 'FinalRepository' # Custom template job has FinalRepository as origin when in the template itself
+
+        # Add the custom job to the non-customized yaml
+        $nonCustomizedYaml.AddCustomJobsToYaml($customTemplateJobs, [CustomizationOrigin]::TemplateRepository)
+
+        $nonCustomizedYaml.content -join "`r`n" | Should -Be ($finalRepositoryYaml.content -join "`r`n")
+
+        # Adding the jobs again doesn't have an effect
+        $nonCustomizedYaml.AddCustomJobsToYaml($customTemplateJobs, [CustomizationOrigin]::TemplateRepository)
+
+        $nonCustomizedYaml.content -join "`r`n" | Should -Be ($finalRepositoryYaml.content -join "`r`n")
+    }
+
+    It('Test YamlClass ApplyTemplateCustomizations') {
+        . (Join-Path $scriptRoot "yamlclass.ps1")
+
+        $srcContent = Get-Content (Join-Path $PSScriptRoot 'YamlSnippet.txt')
+        $resultContent = Get-Content (Join-Path $PSScriptRoot 'CustomizedYamlSnippet-FinalRepository.txt')
+
+        [Yaml]::ApplyTemplateCustomizations([ref] $srcContent, (Join-Path $PSScriptRoot 'CustomizedYamlSnippet-TemplateRepository.txt'))
+
+        $srcContent | Should -Be ($resultContent -join "`n")
+    }
+
+    It('Test YamlClass ApplyFinalCustomizations') {
+        . (Join-Path $scriptRoot "yamlclass.ps1")
+
+        $srcContent = Get-Content (Join-Path $PSScriptRoot 'YamlSnippet.txt')
+        $resultContent = Get-Content (Join-Path $PSScriptRoot 'CustomizedYamlSnippet-TemplateRepository.txt')
+
+        [Yaml]::ApplyFinalCustomizations([ref] $srcContent, (Join-Path $PSScriptRoot 'CustomizedYamlSnippet-TemplateRepository.txt')) # Threat the template repo as a final repo
+
+        $srcContent | Should -Be ($resultContent -join "`n")
+    }
+}
+
+Describe "CheckForUpdates Action: CheckForUpdates.HelperFunctions.ps1" {
```

Comment:
It seems you had a "bad merge" here... Some tests have been duplicated. I think you need to get that out to an external comparison tool to figure out...

Probably my fault from the other PR merged yesterday... Sorry about that! 🫣